scipy/stats/_distn_infrastructure.py

```diff
@@ -3913,6 +3911,105 @@ def _get_support(self, *args):
         """
         return self.a, self.b
 
+    def _pmf(self, x):
+        return rv_sample._pmf(self, x)
+
+    def _cdf(self, x):
+        return rv_sample._cdf(self, x)
+
+    def _ppf(self, q):
+        return rv_sample._ppf(self, q)
+
+    def _rvs(self, size=None, random_state=None):
+        return rv_sample._rvs(self, size, random_state)
+
+    def _entropy(self):
+        return rv_sample._entropy(self)
+
+    def generic_moment(self, n):
+        return rv_sample.generic_moment(self, n)
+
+    def _expect(self, fun, lb, ub, *args, **kwds):
+        return rv_sample._expect(self, fun, lb, ub, *args, **kwds)
+
+
+class rv_sample(rv_discrete):
+    def __init__(self, xk, pk, name=None, longname=None, seed=None):
+        super(rv_discrete, self).__init__(seed)
+
+        # cf generic freeze
+        self._ctor_param = dict(xk=xk, pk=pk, seed=seed,
+                                name=name, longname=longname)
+
+        self.badvalue = np.nan
+        self.vecentropy = self._entropy
+        self.inc = 1
```

Comment:
`self.inc = 1` is not OK :-). This attribute is broken across all discrete distributions, the assumption is baked rather deep down. I once dropped into the test loop in `test_discrete_basic` a subclass with `self.inc=2` and a bunch of tests pass because in many places it really looks into `n+1` or some such.

That said, here this attribute is because of a test in `test_moments` which compares moments to a generic implementation IIRC, and that expects `self.inc` because of 
https://github.com/scipy/scipy/blob/main/scipy/stats/_distn_infrastructure.py#L3750

I did not want to touch tests in the PR (they could be improved, I'm sure)

Reply:
Yeah... let's leave that alone. I don't think we need to touch the tests. I am surprised at myself for even wanting to work on this, but gh-8057 has been nagging me.